ble/UUID.h

```diff
@@ -42,7 +97,7 @@ class UUID {
      *         different service or characteristics on the BLE device.
      *
      * @param longUUID
-     *          The 128-bit (16-byte) UUID value, MSB first (big-endian).
+     *          The 128-bit (16-byte) UUID value, LSB first (little-endian).
```

Comment:
Having a default ordering is why we are in this bind to begin with. Unless we have explicit ordering at construction time people will use them incorrectly.

You can use the byte array for so much more than just full UUID comparisons. You might want to print out the UUID on your debug terminal or put it on a screen or do a partial comparison to see whether UUIDs belong together. And that is just examples I can think of off the top of my head. 

And we still need to cater for stack developers which definitely need ```getBaseUUID```.